libnopegl/src/utils.h

```diff
@@ -60,6 +60,16 @@
 #define NGLI_MAX(a, b) ((a) > (b) ? (a) : (b))
 #define NGLI_CLAMP(x, min, max) NGLI_MAX(NGLI_MIN(x, max), min)
 
+#define NGLI_ARRAY_MEMDUP(dst, src, name) do {                          \
```

Comment:
Not blocking but I feel like instead we could have the `darray` API with an initializer from a data pointer + count.

`ngli_darray_push_n(struct darray *darray, const void *elements, size_t n);`. You would then have a darrays to work with, which is easier to abstract than the pair of `##name` / `nb_##name` fields

Reply:
Indeed, a darray_dup() or darray_copy() is easier to work with but it only works with arrays on the heap. An alternative would be something like:
``` c
#define NGLI_ARRAY(type) struct { type *elems; size_t nb_elems; } and have a bunch of macro associated with it (like `NGLI_ARRAY_MEMDUP()`, `NGLI_ARRAY_APPEND()`, ...)

This has the benefit of keeping the type and you can style do something like this:
```
NGLI_ARRAY(int) array = {.elems = (int[]){1, 2}, .nb_elems = 2};
```
If you don't want to deal with alloc/dealloc.